File: tudushnik/forms/project.py
from django import forms

from tudushnik.models.project import Project


class AddProjectForm(forms.ModelForm):
    class Meta:
        model = Project
        fields = ['title', 'description', 'color', 'tags']
        widgets = {
            'title': forms.TextInput(attrs={
                'class': 'form-input'
            }),
            'description': forms.Textarea(attrs={
                'cols': 60,
                'rows': 10
            }),
            'color': forms.TextInput(attrs={
                'type': 'color'
            }),
            'tags': forms.SelectMultiple(),
        }
        labels = {
            'title': 'Название',
            'description': 'Описание',
            'color': 'Цвет',
            'tags': 'Тэги'
        }


class ProjectUpdateForm(forms.ModelForm):
    class Meta:
        model = Project
        fields = ['title', 'description', 'color', 'tags']
        widgets = {
            'title': forms.TextInput(attrs={
                'class': 'form-input'
            }),
            'description': forms.Textarea(attrs={
                'cols': 60,
                'rows': 10
            }),
            'color': forms.TextInput(attrs={
                'type': 'color'
            }),
            'tags': forms.SelectMultiple(),
        }
        labels = {
            'title': 'Название',
            'description': 'Описание',
            'color': 'Цвет',
            'tags': 'Тэги'
        }

File: tudushnik/static/tudushnik/js/src/project/project.js
class Project{
    #app
    #pk;
    #title;
    #description;
    #color;
    #tags;

    constructor(pk, title, description, color, tags){
        this.#app = null
        this.#pk = pk
        this.#title = title
        this.#description = description
        this.#color = color
        this.#tags = tags
    }
    set app(value){
        this.#app = value
    }
    get app(){
        return this.#app;
    }
    get pk(){
        return this.#pk;
    }
    get title(){
        return this.#title;
    }
    get description(){
        return this.#description;
    }
    get color(){
        return this.#color;
    }
    get tags(){
        return this.#tags;
    }

}

export {
    Project
}